runtime/catalog-karapace/src/main/java/io/aklivity/zilla/runtime/catalog/karapace/internal/CachedSchema.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2021-2023 Aklivity Inc
+ *
+ * Licensed under the Aklivity Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ *   https://www.aklivity.io/aklivity-community-license/
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package io.aklivity.zilla.runtime.catalog.karapace.internal;
+
+import java.util.concurrent.atomic.AtomicInteger;
+
+public class CachedSchema
+{
+    public static final String SCHEMA_PLACEHOLDER = "schema";
+    public static final CachedSchema IN_PROGRESS = new CachedSchema(SCHEMA_PLACEHOLDER, new AtomicInteger(0));
+
+    public String schema;
+    public AtomicInteger event;
+
+    public CachedSchema(
+        String schema,
+        AtomicInteger event)
+    {
+        this.schema = schema;
+        this.event = event;
+    }
```

Comment:
Does it make sense to have a convenience constructor with just `schema` defauting to `new AtomicInteger()`.